data_management/models.py

```diff
@@ -536,12 +522,13 @@ class StorageLocation(BaseModel):
 
     path = models.CharField(max_length=PATH_FIELD_LENGTH, null=False, blank=False)
     hash = models.CharField(max_length=CHAR_FIELD_LENGTH, null=False, blank=False)
+    public = models.BooleanField(default=True)
     storage_root = models.ForeignKey(StorageRoot, on_delete=models.CASCADE, related_name='locations')
 
     class Meta:
         constraints = [
             models.UniqueConstraint(
-                fields=('storage_root', 'path', 'hash'),
+                fields=('storage_root', 'path', 'hash', 'public'),
```

Comment:
Does `path` (or even `storage_root`?) need to be part of the uniqueness constraint here? If it's the same file and the same `public`-ness is that enough?

Having said that, we have a problem if the file is only remote and we want to create a local copy, then that may be impossible... we could fix that by adding `storage_root` too, or we could create an automated field - `local` - that is inherited from the StorageRoot, and add that as a uniqueness constraint.

Does any of that make sense?

Reply:
Probably `path` can be dropped from the uniqueness constraint. Keeping `storage_root` would allow a single registry to reference the same file in different places (storage_roots).

Reply:
That seems reasonable to me. Go for it.